cluster/gce/util.sh

```diff
@@ -794,14 +794,11 @@ function construct-linux-kubelet-flags {
     flags+=" --register-with-taints=${NODE_TAINTS}"
   fi
 
-  flags+=" --container-runtime=remote"
-  if [[ "${CONTAINER_RUNTIME}" == "containerd" ]]; then
-    CONTAINER_RUNTIME_ENDPOINT=${KUBE_CONTAINER_RUNTIME_ENDPOINT:-unix:///run/containerd/containerd.sock}
-    flags+=" --runtime-cgroups=/system.slice/containerd.service"
-  fi
+  CONTAINER_RUNTIME_ENDPOINT=${KUBE_CONTAINER_RUNTIME_ENDPOINT:-unix:///run/containerd/containerd.sock}
+  flags+=" --container-runtime-endpoint=${CONTAINER_RUNTIME_ENDPOINT}"
 
-  if [[ -n "${CONTAINER_RUNTIME_ENDPOINT:-}" ]]; then
-    flags+=" --container-runtime-endpoint=${CONTAINER_RUNTIME_ENDPOINT}"
+  if [[ "${CONTAINER_RUNTIME_ENDPOINT}" =~ /containerd.sock$ ]]; then
+    flags+=" --runtime-cgroups=/system.slice/containerd.service"
```

Comment:
@adisky This is The only place where some containerd specific logic was needed.
Everything else seems to be based mostly on `KUBE_CONTAINER_RUNTIME_ENDPOINT` and `KUBE_CONTAINER_RUNTIME_NAME`.